### PR TITLE
handle sprockets > 4 does not chain processors anymore

### PR DESCRIPTION
**Pull request by jklimke**
Added a ERB Preprocessor to webmanifests in order to reenable webmanifest rendering

### CODE REVIEW

File: lib/generators/web_app_manifest_initializer.txt [hunk 1]
```diff
@@ -6,5 +6,10 @@
 # to asset_path in the _favicon.html.erb partial.
 
 Rails.application.config.assets.configure do |env|
-  env.register_mime_type('application/manifest+json', extensions: ['.webmanifest'])
+  if(Sprockets::VERSION.split('.')[0].to_i >= 4)
```
Comment (KevinBongart) [inline lib/generators/web_app_manifest_initializer.txt]: No need for the parenthesis, and Ruby usually favors spaces around conditions:

```suggestion
  if Sprockets::VERSION.split('.')[0].to_i >= 4
```

Also works:

```ruby
if Sprockets::VERSION.to_i >= 4
```

Also, do you know if there's a more specific way to determine why adding the `.erb` extension is needed, beyond the version itself? Maybe a Sprocket internal method or feature flag?

**Comment by KevinBongart** [comment]
@phbernard Would be great to merge this PR, as Sprockets 4 is getting traction

File: lib/generators/web_app_manifest_initializer.txt (hunk 1 above)
Comment (jklimke) [reply lib/generators/web_app_manifest_initializer.txt]: At least i am not aware of any flags indicting that the processors are not chained automatically anymore ([see here](https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#adding-erb-support-to-your-extension)). They just don't support this chaining by file ending anymore.

Comment (KevinBongart) [reply lib/generators/web_app_manifest_initializer.txt]: Got it. In this case, here's the same approach but with a bit less repetition, which I think highlights the different behaviors between Sprockets version a bit more:

```ruby
mime_type = 'application/manifest+json'
extensions = ['.webmanifest']

if Sprockets::VERSION.to_i >= 4
  extensions << '.webmanifest.erb'
  env.register_preprocessor(mime_type, Sprockets::ERBProcessor)
end

env.register_mime_type(mime_type, extensions: extensions)
```

What do you think?

Regardless, your PR works, is readable and should go in :) Thanks for the fix

Comment (jklimke) [reply lib/generators/web_app_manifest_initializer.txt]: Thanks for the review. I adapted the initializer code. 

Nevertheless, the fix will only work for newly installed rails_real_favcon installations as the existing files aren't patched. Maybe there should be another hint of how to handle sprockets 4 within the readme or changelog when a new version is published.

File: lib/generators/web_app_manifest_initializer.txt [hunk 2]
```diff
@@ -6,5 +6,13 @@
 # to asset_path in the _favicon.html.erb partial.
 
 Rails.application.config.assets.configure do |env|
-  env.register_mime_type('application/manifest+json', extensions: ['.webmanifest'])
+  mime_type = 'application/manifest+json'
+  extensions = ['.webmanifest']  
+
+  if Sprockets::VERSION.to_i >= 4
+    extensions << '.webmanifest.erb'    
+    env.register_preprocessor('application/manifest+json', Sprockets::ERBProcessor)
```
Comment (KevinBongart) [inline lib/generators/web_app_manifest_initializer.txt]:
```suggestion
    env.register_preprocessor(mime_type, Sprockets::ERBProcessor)
```

File: lib/generators/web_app_manifest_initializer.txt [hunk 3]
```diff
@@ -6,5 +6,13 @@
...
+  end  
+  
+  env.register_mime_type('application/manifest+json', extensions: extensions)
```
Comment (KevinBongart) [inline lib/generators/web_app_manifest_initializer.txt]:
```suggestion
  env.register_mime_type(mime_type, extensions: extensions)
```

**Comment by phbernard** [comment]
Finally merged and published a minute ago (v0.1.0). Thank you @jklimke and @KevinBongart for your comprehensive analysis and code!

**Pull request merged**